File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int res=0;                            // Global ariable lelo ; usko shuru mei se hi 0 se init karo
void fun(TreeNode*root,int sum){
    // agar root hi nhi hai toh tree hi nhi hai 
    if(root==nullptr){
        return;
    }

    sum=sum*10+root->val;                    // path k along jodtey jao uskey liye 

    // agar leaf node hai 
    if(root->left==nullptr && root->right==nullptr){
        res+=sum;                                // sum jo bhi aaega usko res mein uopdate kro 
        return;
    }

    // Agar Non leaf node hai ; toh left and right ki orr extend krtey chlo 
    fun(root->left,sum);                  // left mei chlo 
    fun(root->right,sum);                 //right mei chloo

    return; 
}

    int sumNumbers(TreeNode* root) {
        // recursive Function call kardo 
        fun(root,0);

        return res;
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int res=0;                             // shuru mei 0 se hi lelo res wla 
    public void fun(TreeNode root,int sum){
        // agar tree ka root hi null hai; yni tree hi nhi hai 
        if(root==null){
            return;
        }

        sum=sum*10+root.val;               // jod loo path k along wley nodes k data se root ko bhi 

        // Agar leaf node tak aagya 
        if(root.left==null && root.right==null){
            res+=sum;                       // res ko update krdo sum se 
            return;
        }

        // Agar Non leaf mei hi ho ; tab extend kro left and right mei 
        fun(root.left,sum);
        fun(root.right,sum);

        return;
    }

    public int sumNumbers(TreeNode root) {
        // Recursive Function call kardo 
        fun(root,0);

        return res; 
    }
}